Run as non-root

iaindillingham opened the pull request:
Closes #196

madwort commented on docker/Dockerfile:
```diff
@@ -154,6 +154,9 @@ LABEL org.opencontainers.image.created=$BUILD_DATE
 ARG GITREF=unknown
 LABEL org.opencontainers.image.revision=$GITREF
 
+ARG USERID=10001
+ARG GROUPID=10001
```
hmm, so `10001` is the user id for job-server, defined [here](https://github.com/ebmdatalab/sysadmin/blob/011eda3ae139da08cd7259a06ba78f9790cfae90/infra/job-server/tasks.yaml#L12-L23)

reports uses `10002` defined [here](https://github.com/ebmdatalab/sysadmin/blob/011eda3ae139da08cd7259a06ba78f9790cfae90/infra/reports/tasks.yaml)

not sure which uid we should use for this.

madwort replied:
actually, looking at `dokku3:/etc/passwd` we don't have a `10001` on that server, so we should change this. 

```
reports:x:10002:10002:Dedicated user for reports service to run as:/home/reports:/bin/sh
opencodelists:x:10003:10003:Dedicated user for opencodelists service to run as:/home/opencodelists:/bin/sh
ebmbot:x:10004:10004:Dedicated user ebmbot for bennettbot service to run as:/home/ebmbot:/bin/sh
```

madwort replied:
see also https://github.com/ebmdatalab/sysadmin/pull/308/files

rebkwok replied:
See https://github.com/ebmdatalab/sysadmin/pull/311
When this is merged, you'll just need to add an extra set of items for this user

iaindillingham replied:
I've added a user/group for actions-registry on dokku3 and updated `docker/Dockerfile`. See ebmdatalab/sysadmin#322.

rebkwok commented:
Does actions-registry write anything (other than the db, which I assume is in a mounted volume in the dokku app)?  This was the thing that caused most problems for opencodelists. If everything it needs write access to is in a mounted storage volume, the sysadmin ansible playbook will do the work of making sure the non-root user owns it.

iaindillingham commented:
Thanks for your comments, @madwort and @rebkwok. I've added a user/group for actions-registry on dokku3 and updated `docker/Dockerfile`. See ebmdatalab/sysadmin#322.

> Does actions-registry write anything?

Other than the DB, no, it doesn't.

rebkwok commented on docker/Dockerfile:
```diff
@@ -154,6 +154,9 @@ LABEL org.opencontainers.image.created=$BUILD_DATE
 LABEL org.opencontainers.image.revision=$GITREF
 
+ARG USERID=10004
+ARG GROUPID=10004
```
Sorry, this uid is taken (by ebmbot); 10005 should be OK

iaindillingham replied:
Thanks, @rebkwok. I've updated to 10005.
